Cannot compile on OS X 10.14.2

Compilation dies here:

_make[2]: Entering directory `/opt/local/var/macports/build/_Users_alien_ports_emulators_opencbm/opencbm/work/code/xu1541/lib'
cc -I/opt/local/include -I/opt/local/include/ -g -Wall -pedantic -I../include/   -c -o close.o close.c
In file included from close.c:1:
../include/xu1541lib.h:57:8: error: unknown type name 'libusb_device_handle'
extern libusb_device_handle * xu1541lib_find(void);
       ^
../include/xu1541lib.h:58:8: error: unknown type name 'libusb_device_handle'
extern libusb_device_handle * xu1541lib_find_in_bootmode(unsigned int *p_soft_bootloader_mode);
       ^
../include/xu1541lib.h:59:47: error: unknown type name 'libusb_device_handle'
extern void                   xu1541lib_close(libusb_device_handle *handle);
                                              ^
../include/xu1541lib.h:60:57: error: unknown type name 'libusb_device_handle'
extern int                    xu1541lib_get_device_info(libusb_device_handle *handle, xu1541_device_info_t *device_info, unsigned int device_info_length);
                                                        ^
../include/xu1541lib.h:61:61: error: unknown type name 'libusb_device_handle'
extern void                   xu1541lib_display_device_info(libusb_device_handle *handle);
                                                            ^
../include/xu1541lib.h:62:63: error: unknown type name 'libusb_device_handle'
extern int                    xu1541lib_is_in_bootloader_mode(libusb_device_handle *handle);
                                                              ^
../include/xu1541lib.h:63:46: error: unknown type name 'libusb_device_handle'
extern void                   xu1541lib_wait(libusb_device_handle *handle);
                                             ^
../include/xu1541lib.h:64:58: error: unknown type name 'libusb_device_handle'
extern int                    xu1541lib_set_to_boot_mode(libusb_device_handle *handle);
                                                         ^
../include/xu1541lib.h:65:54: error: unknown type name 'libusb_device_handle'
extern int                    xu1541lib_get_pagesize(libusb_device_handle *handle);
                                                     ^
close.c:9:22: error: unknown type name 'libusb_device_handle'
void xu1541lib_close(libusb_device_handle *handle) {
                     ^
close.c:11:13: warning: implicit declaration of function 'libusb_release_interface' is invalid in C99 [-Wimplicit-function-declaration]
  int ret = libusb_release_interface(handle, 0);
            ^
close.c:12:14: error: use of undeclared identifier 'LIBUSB_SUCCESS'
  if (ret != LIBUSB_SUCCESS) {
             ^
close.c:13:40: warning: implicit declaration of function 'libusb_error_name' is invalid in C99 [-Wimplicit-function-declaration]
    fprintf(stderr, "USB error: %s\n", libusb_error_name(ret));
                                       ^
close.c:13:40: warning: format specifies type 'char *' but the argument has type 'int' [-Wformat]
    fprintf(stderr, "USB error: %s\n", libusb_error_name(ret));
                                ~~     ^~~~~~~~~~~~~~~~~~~~~~
                                %d
close.c:17:3: warning: implicit declaration of function 'libusb_close' is invalid in C99 [-Wimplicit-function-declaration]
  libusb_close(handle);
  ^
4 warnings and 11 errors generated._

The issue has to do something with the fact that wrong/no headers are included.

"libusb_device_handle" function is found in "/opt/local/include/libusb-1.0/libusb.h" but NOT in "/opt/local/include/usb.h"

As you can see "cc..." is totally missing the -DHAVE_LIBUSB=1 -DHAVE_LIBUSB1=1 -DHAVE_LIBUSB_1_0=1 stuff....thus "../include/xu1541lib.h" does not include anything:

#if HAVE_LIBUSB0
#include <usb.h>
#elif HAVE_LIBUSB1
#include <libusb.h>
#endif

--- a/opencbm/ports/MacPorts/Portfile
+++ b/opencbm/ports/MacPorts/Portfile
@@ -19,8 +19,9 @@ long_description    The opencbm (cbm4linux) package contains user space \
 platforms           darwin
 homepage            https://sourceforge.net/projects/opencbm/
 
-depends_build       port:cc65
-depends_lib         port:libusb-compat
+depends_build       port:cc65 \
+                    port:pkgconfig
+depends_lib         port:libusb
 
 fetch.type          git
 git.url             https://github.com/OpenCBM/OpenCBM.git

--- a/xu1541/include/common.mk
+++ b/xu1541/include/common.mk
@@ -31,51 +31,42 @@ else
  CFLAGS_EXTRA =
  # determine libusb location
  OS=$(shell uname -s)
- ifeq "$(OS)" "Darwin"
-   # MacOS compilation:
+ ifeq "$(shell uname -o)" "Cygwin"
+   # Cygwin compilation
 
-   LIBUSB_DIR = $(shell libusb-config --prefix)
-   LDFLAGS_EXTRA = $(shell libusb-config --libs)
-   CFLAGS_EXTRA = $(shell libusb-config --cflags)
+   # Tell, where your libusb-win32 installation resides
+   # (if not part of Cygwin in the latest version)
+   #
+   # LIBUSB_DIR=/cygdrive/c/drivers/LibUSB
+#   LIBUSB_DIR=/cygdrive/n/Programme/LibUSB
+   LIBUSB_DIR=$(HOME)/LibUSB
+   LIB_WIN = .lib
+   EXE_SUFFIX = .exe
 
+   LDFLAGS_EXTRA=-mno-cygwin -L $(LIBUSB_DIR)/lib/gcc
+   CYGWIN=1
+
+   # For compiling for MinGw32 target
+   #
+   CFLAGS_EXTRA=-mno-cygwin -DWIN
  else
-   ifeq "$(shell uname -o)" "Cygwin"
-     # Cygwin compilation
 
-     # Tell, where your libusb-win32 installation resides
-     # (if not part of Cygwin in the latest version)
-     #
-     # LIBUSB_DIR=/cygdrive/c/drivers/LibUSB
-#     LIBUSB_DIR=/cygdrive/n/Programme/LibUSB
-     LIBUSB_DIR=$(HOME)/LibUSB
-     LIB_WIN = .lib
-     EXE_SUFFIX = .exe
+   ### LIBUSB_DIR is a dummy, not needed here
+   LIBUSB_DIR=../
 
-     LDFLAGS_EXTRA=-mno-cygwin -L $(LIBUSB_DIR)/lib/gcc
-     CYGWIN=1
+   HAVE_LIBUSB0 = ${shell pkg-config libusb && echo 1}
+   HAVE_LIBUSB1 = ${shell pkg-config libusb-1.0 && echo 1}
 
-     # For compiling for MinGw32 target
-     #
-     CFLAGS_EXTRA=-mno-cygwin -DWIN
-   else
+   ifneq ($(strip $(HAVE_LIBUSB0)),)
+     HAVE_LIBUSB=1
+     CFLAGS_EXTRA=-DHAVE_LIBUSB=1 -DHAVE_LIBUSB0=1 $(shell pkg-config --cflags libusb)
+     LDFLAGS_EXTRA=$(shell pkg-config --libs libusb)
+   endif
 
-     ### LIBUSB_DIR is a dummy, not needed here
-     LIBUSB_DIR=../
-
-     HAVE_LIBUSB0 = ${shell pkg-config libusb && echo 1}
-     HAVE_LIBUSB1 = ${shell pkg-config libusb-1.0 && echo 1}
-
-     ifneq ($(strip $(HAVE_LIBUSB0)),)
-       HAVE_LIBUSB=1
-       CFLAGS_EXTRA=-DHAVE_LIBUSB=1 -DHAVE_LIBUSB0=1 $(shell pkg-config --cflags libusb)
-       LDFLAGS_EXTRA=$(shell pkg-config --libs libusb)
-     endif
-
-     ifneq ($(strip $(HAVE_LIBUSB1)),)
-       HAVE_LIBUSB=1
-       CFLAGS_EXTRA=-DHAVE_LIBUSB=1 -DHAVE_LIBUSB1=1 -DHAVE_LIBUSB_1_0=1 $(shell pkg-config --cflags libusb-1.0)
-       LDFLAGS_EXTRA=$(shell pkg-config --libs libusb-1.0)
-     endif
+   ifneq ($(strip $(HAVE_LIBUSB1)),)
+     HAVE_LIBUSB=1
+     CFLAGS_EXTRA=-DHAVE_LIBUSB=1 -DHAVE_LIBUSB1=1 -DHAVE_LIBUSB_1_0=1 $(shell pkg-config --cflags libusb-1.0)
+     LDFLAGS_EXTRA=$(shell pkg-config --libs libusb-1.0)
    endif
  endif
 endif

